overlord/devicestate/devicestate.go

```diff
@@ -1231,6 +1238,20 @@ func checkForInvalidSnapsInModel(model *asserts.Model, vSets *snapasserts.Valida
 	return nil
 }
 
+func checkForSystemSeed(st *state.State, deviceCtx snapstate.DeviceContext) (bool, error) {
+	// on non-classic systems, we will always have a seed partition
+	if !deviceCtx.Classic() {
+		return true, nil
+	}
```

Comment:
This is true, but `HasRole` is not really that costly and it will return the right answer anyway, so maybe this could be removed to simplify the function a bit.

Reply:
This was an easy way to prevent updating a lot of tests :slightly_smiling_face:. I'm happy to change it though, if you think it is best.

Reply:
Oh I see - ok, I think it is fine but please add a comment/TODO about this